services/QuillLMS/engines/evidence/app/controllers/evidence/research/gen_ai/datasets_controller.rb

```diff
@@ -9,25 +9,42 @@ def new
         end
 
         def create
-          @dataset = stem_vault.datasets.new(dataset_params)
-
-          if @dataset.save
-            DatasetImporter.run(dataset: @dataset, file:)
-            redirect_to @dataset
-          else
-            render :new
+          if data_subset?
```

Comment:
Take or leave, but considering how different the create paths (branching logic, different params) are, you might consider having a `data_subsets_controller` with just a `create` method. It could be nested with in the data_set routes also (but again, take or leave).

Reply:
I think if these constructions start to grow, I will extract into some separate controller.